registry/client/repository.go

```diff
@@ -576,90 +572,93 @@ func (bs *blobs) Put(ctx context.Context, mediaType string, p []byte) (distribut
 	return writer.Commit(ctx, desc)
 }
 
-func (bs *blobs) Create(ctx context.Context) (distribution.BlobWriter, error) {
-	bs.cacheLock.Lock()
-	if bs.cachedBlobUpload != nil {
-		upload := bs.cachedBlobUpload
-		bs.cachedBlobUpload = nil
-		bs.cacheLock.Unlock()
-
-		return upload, nil
+type createOptions struct {
+	mount struct {
+		shouldMount bool
+		from        reference.Canonical
 	}
-	bs.cacheLock.Unlock()
+}
 
-	u, err := bs.ub.BuildBlobUploadURL(bs.name)
-	if err != nil {
-		return nil, err
-	}
+type optionFunc func(interface{}) error
```

Comment:
is optionFunc necessary?  Can `Apply' be a member of `createOptions` and `WithMountFrom` just return a new `createOptions` ?